controllers/machinedeployment_controller.go

```diff
@@ -213,6 +236,80 @@ func (r *MachineDeploymentReconciler) reconcile(ctx context.Context, cluster *cl
 	return ctrl.Result{}, errors.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
+func (r *MachineDeploymentReconciler) rolloutAfter(ctx context.Context, machineDeployment *clusterv1.MachineDeployment, newMachineSet *clusterv1.MachineSet) error {
+	var machinesToRollout []*clusterv1.Machine
+	machines, err := r.getMachinesForMachineSet(ctx, newMachineSet)
+	if err != nil {
+		return err
+	}
+
+	// Find Machines needing to rollout after.
+	for key := range machines {
+		if machines[key].DeletionTimestamp.IsZero() &&
+			machines[key].CreationTimestamp.Before(machineDeployment.Spec.RolloutAfter) {
```

Comment:
Maybe I missed it, but don't we also have to check that time.Now() > RolloutAfter ? (in KCP this is done indirectly by comparing against `reconciliationTime`)